provider/azure/environ.go

```diff
@@ -1672,7 +1672,7 @@ func (env *azureEnviron) deleteControllerManagedResourceGroups(ctx context.Provi
 
 	// Walk all the pages of results so we can get a total list of groups to remove.
 	var groupNames []*string
-	for ; result.NotDone(); err = result.NextWithContext(sdkCtx) {
+	for ; result.NotDone(); result.NextWithContext(sdkCtx) {
```

Comment:
I think we need to check this error as it appears that NextWithContext can do paging (as in make paginated API calls)

Reply:
In fact, there is possibly infinite retry logic going on here.

Reply:
+1, should not ignore the error as the code currently does